### PR TITLE
Retrieve user's saved pay methods (fixes #37)

### LINKED ISSUE #37: Create end point to list existing cards
To suport mozilla/payments#65 we'll need a way to query payments-service and return a list of existing payment methods.

### PR DESCRIPTION
**Pull request by kumar303**

### CODE REVIEW

File: payments_service/braintree/views.py [hunk 1]
```diff
@@ -24,6 +24,29 @@ class TokenGenerator(SolitudeBodyguard):
...
+        This sets the right default parameters that we
+        want to send to Solitude, allowing for some overrides.
+        The important part is that it only lets you get
+        payment methods for the logged in user.
+        """
+        replaced_kw = {
+            'active': kw.get('active', 1),  # active by default
+            'braintree_buyer__buyer__uuid': request.user.uuid,
+        }
+        return tuple(), replaced_kw
```
Comment (andymckay) [inline payments_service/braintree/views.py:47]: not return args?

File: payments_service/solitude/__init__.py [hunk 2]
```diff
@@ -37,15 +37,36 @@ class SolitudeBodyguard(APIView):
     # resource = 'services.status'
     resource = None
 
-    def _api_request(self, method, *args, **kw):
+    def replace_call_args(self, django_request, args, kw):
```
Comment (andymckay) [inline payments_service/solitude/__init__.py:40]: Maybe a note about `django_request`, actually surprised flake8 doesn't complain about it be defined but never used or returns. Oh man if only we had interfaces.

File: payments_service/solitude/tests/test.py [hunk 3]
```diff
@@ -69,6 +76,32 @@ def test_pass_post_params_to_solitude(self):
         eq_(call_args[0][0]['foo'], params['foo'])
         eq_(call_args[0][0]['bar'], params['bar'])
 
+    def test_pass_get_params_to_solitude(self):
+        params = {'foo': '1', 'bar': 'baz'}
+        self.get(query_params=params)
+        assert self.solitude.services.status.get.called
```
Comment (andymckay) [inline payments_service/solitude/tests/test.py]: I think you can remove this line because if it didn't get called the next assertion would fail.

**Comment by andymckay** [comment]
Probably don't need to worry about it now, but I'm curious what happens with pagination here. If you have more than 20 cards, it will start to paginate. Perhaps we should limit the number of active cards to make life easier?

**Comment by andymckay** [comment]
r+wc

File: payments_service/braintree/views.py (hunk 1 above)
Comment (kumar303) [inline payments_service/braintree/views.py:47]: I think there could be some security issues with allowing args. For example, `braintree.mozilla.paymethod.get(another_users_pk)` might be possible.

File: payments_service/solitude/__init__.py (hunk 2 above)
Comment (kumar303) [inline payments_service/solitude/__init__.py:40]: all roads lead to Go

Comment (kumar303) [inline payments_service/solitude/__init__.py:40]: I seem to recall suggesting we write the service in Go :)

**Comment by kumar303** [comment]
Mmm, pagination. I guess we sort of got that for free in solitude via tasty pie? I wonder if the DRF model view adds anything by default there.

**Comment by andymckay** [comment]
DRF does pagination for you, it looks just like Tastypies. Going to be a pain.

**Pull request merged**